plugin/trino-druid/src/test/java/io/trino/plugin/druid/DruidCreateAndInsertDataSetup.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.druid;
+
+import io.airlift.log.Logger;
+import io.trino.plugin.druid.ingestion.IndexTaskBuilder;
+import io.trino.plugin.druid.ingestion.TimestampSpec;
+import io.trino.testing.datatype.ColumnSetup;
+import io.trino.testing.datatype.DataSetup;
+import io.trino.testing.sql.TemporaryRelation;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static java.lang.String.format;
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+public class DruidCreateAndInsertDataSetup
+        implements DataSetup
+{
+    private static final Logger log = Logger.get(DruidCreateAndInsertDataSetup.class);
+    private final TestingDruidServer druidServer;
+    private final String dataSourceNamePrefix;
+
+    public DruidCreateAndInsertDataSetup(TestingDruidServer druidServer, String dataSourceNamePrefix)
+    {
+        this.druidServer = druidServer;
+        this.dataSourceNamePrefix = dataSourceNamePrefix;
+    }
+
+    @Override
+    public TemporaryRelation setupTemporaryRelation(List<ColumnSetup> inputs)
+    {
+        DruidTable testTable = new DruidTable(this.dataSourceNamePrefix);
+        try {
+            ingestData(testTable, inputs);
+        }
+        catch (Exception e) {
```

Comment:
Why it is so generic? Can you have interrupted exception here?

Reply:
TestingDruidServer uses `Thread#sleep` for retrying when we it looks if the tables exists or not. We can replace it with `Failsafe` in the future. I think it can be addressed in a different PR.